helpers.go

```diff
@@ -445,14 +417,73 @@ func SortedMapValues[T any](m map[string]T) []T {
 	return out
 }
 
+// MergeDependencies merges two sets of package dependencies, a base and a target.
+// If a dependency is set in both, the one from `target` is used, otherwise, the dependency from parent is used.
+// MergeDependencies(nil, child) = child, MergeDependencies(parent, nil) = parent
```

Comment:
```suggestion
// MergeDependencies(nil, child) = child, MergeDependencies(parent, nil) = parent, MergeDependencies(parent, child) = child
```

Is this correct too?

Reply:
Assuming all **fields** of child are non-nil then yes, otherwise some fields may be taken from the parent.